eclipse-che/che-operator
# Configure DWO to not process proxy settings for workspaces Che manages

### What does this PR do?
If Che detects proxy settings on a cluster (either through configuration or the OpenShift cluster-wide proxy), configure the DevWorkspace Operator to not use any proxy settings internally for workspaces. 

This is necessary to allow the `proxy-settings` configmap provisioned by Che to actually be used in workspaces; otherwise, DWO will set the proxy environment variables in the workspace deployment directly, overriding env vars set from the configmap.

In addition, this PR updates the proxy env configmap to include lowercase versions of environment variables (i.e. `http_proxy`, `https_proxy`, `no_proxy`) to try to ensure compatibility with all tools (e.g. `curl` _only_ uses `http_proxy`, not `HTTP_PROXY`)

~This PR is marked WIP as it depends on changes from https://github.com/devfile/devworkspace-operator/pull/1147; once DWO v0.23 is released, the first commit in this PR needs to be updated to use 0.23 instead of a main-branch commit.~

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
Closes https://github.com/eclipse/che/issues/22370

### How to test this PR?
Changes from this PR are pushed to `quay.io/amisevsk/che-operator:proxy-config`

To test, 
1. Start an OpenShift cluster with a cluster-wide proxy configured
2. Install Che + DWO: `chectl server:deploy -p openshift --olm-channel next`
3. Update the eclipse-che CSV (`eclipse-che.<ver>.next`) to use `quay.io/amisevsk/che-operator:proxy-config` for the Che Operator image
4. Start a workspace

Once workspace is running, verify
1. DevWorkspace operator config is configured to not set proxy variables:
    ```
    oc get dwoc -n eclipse-che devworkspace-config -o yaml
    (...)
      config:
        routing:
          proxyConfig:
            httpProxy: ""
            httpsProxy: ""
            noProxy: ""
    (...)
    ```
2. The workspace's deployment does not set proxy environment variables directly
3. Proxy variables are still present in workspace, mounted from configmap
    ```
    $ env | grep -i proxy
    HTTP_PROXY=http://<snip>@10.0.15.91:3128/
    https_proxy=http://<snip>@10.0.15.91:3128/
    http_proxy=http://<snip>@10.0.15.91:3128/
    no_proxy=.cluster.local,#etc
    NO_PROXY=.cluster.local,#etc
    HTTPS_PROXY=http://<snip>@10.0.15.91:3128/
    ```
4. Changes to proxy settings in CheCluster propagate down to workspaces and aren't ignored due to DWO's conflicting config

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
